foundation/src/main/java/org/researchstack/foundation/components/survey/ui/body/FormBody.java

```diff
@@ -66,6 +65,7 @@ public StepResult getStepResult(boolean skipped) {
                 result.setResultForIdentifier(childResult.getIdentifier(), childResult);
             }
         }
+        result.updateEndDate();
```

Comment:
Hmm, my comment seems to have disappeared.

@jdkizer9: I think @albf found a bug

When childResult is not null and contains an endDate, should we use that or the current time as this result's endDate?

Reply:
Yeah, it does look like we aren't updating endDate for form steps. I'm trying to think through edge cases. If someone completes a form step, then goes back, would we want the original completion date or a new one? It looks like all other step endDates are set during getStepResult (via setResult). If that's the case, then for consistency the endDate should probably be the last time the step was viewed. Regarding using child step results, those endDates are (with the exception of TextQuestionBody) set during the call to getStepResult. Therefore, I'd argue that there's really no value in using the latest child result. 

Also, looking over the loop here, it seems like the child step results will always be non-null, even when skipped (at least for bodies defined in Foundation). When skipping the step, child steps return a step result containing a null value. What do you think proper behavior for a skipped Form step should be? Should we return a step result that contains a map of step results, each of which contains a null value? Or should we just return a step result container with an empty map? In the former case, if the child result does return null (in the case of custom form item bodies), we should probably be setting clearing that value in the map.